codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/WaitersGenerator.java

```diff
@@ -300,4 +319,168 @@ public Void visitUnknown(Matcher.UnknownMember unknown) {
+
+        @Override
+        public JmespathExpression visitMultiSelectList(MultiSelectListExpression expression) {
+            return new MultiSelectListExpression(
+                    expression.getExpressions().stream()
+                            .map((e) -> e.accept(this)).collect(Collectors.toList()),
+                    expression.getLine(),
+                    expression.getColumn()
+            );
+        }
+
+        @Override
+        public JmespathExpression visitMultiSelectHash(MultiSelectHashExpression expression) {
+            return expression; // TODO
+        }
+
+        @Override
+        public JmespathExpression visitAnd(AndExpression expression) {
+            return new AndExpression(
+                    expression.getLeft().accept(this),
+                    expression.getRight().accept(this),
+                    expression.getLine(),
+                    expression.getColumn()
+            );
+        }
+
+        @Override
+        public JmespathExpression visitOr(OrExpression expression) {
+            return new OrExpression(
+                    expression.getLeft().accept(this),
+                    expression.getRight().accept(this),
+                    expression.getLine(),
+                    expression.getColumn()
+            );
+        }
+
+        @Override
+        public JmespathExpression visitNot(NotExpression expression) {
+            return new NotExpression(
+                    expression.getExpression().accept(this),
+                    expression.getLine(),
+                    expression.getColumn()
+            );
+        }
+
+        @Override
+        public JmespathExpression visitProjection(ProjectionExpression expression) {
+            return new ProjectionExpression(
+                    expression.getLeft().accept(this),
+                    expression.getRight().accept(this),
+                    expression.getLine(),
+                    expression.getColumn()
+            );
+        }
+
+        @Override
+        public JmespathExpression visitFilterProjection(FilterProjectionExpression expression) {
+            return new FilterProjectionExpression(
+                    expression.getLeft().accept(this),
+                    expression.getComparison().accept(this),
+                    expression.getRight().accept(this),
+                    expression.getLine(),
+                    expression.getColumn()
+            );
+        }
+
+        @Override
+        public JmespathExpression visitObjectProjection(ObjectProjectionExpression expression) {
+            return new ObjectProjectionExpression(
+                    expression.getLeft().accept(this),
+                    expression.getRight().accept(this),
+                    expression.getLine(),
+                    expression.getColumn()
+            );
+        }
+
+        @Override
+        public JmespathExpression visitSlice(SliceExpression expression) {
+            return expression; // TODO
```

Comment:
Are these optional or mandatory for us?

Reply:
This one - I'm on the fence about.  Theres nothing that gets transformed in this, but we shouldn't re-use the existing expression object (all of the visitors need to return a copy).